### ISSUE TITLE
0.7 breaks on upgrade from 0.6

### ISSUE THREAD

**Issue by maletor** [mocleiri/github-oauth-plugin#9]
Jenkins throws a 404 with a null in the URL when I upgraded the plugin to 0.7.

**Comment by vkravets** [comment]
Issue is related to ability to specify github uri for enterprise github hosting. To resolve this issue need to edit config.xml in jenkins home directory. Find:

``` xml
  <securityRealm class="org.jenkinsci.plugins.GithubSecurityRealm">
    <clientID>your clientID</clientID>
    <clientSecret>your cientServer</clientSecret>
  </securityRealm>
```

and change to:

``` xml
  <securityRealm class="org.jenkinsci.plugins.GithubSecurityRealm">
    <githubUri>https://github.com</githubUri>
    <clientID>your clientID</clientID>
    <clientSecret>your cientServer</clientSecret>
  </securityRealm>
```

After it everything is working well.

**Comment by maletor** [comment]
Ah great. Thanks!

I'll leave it up to the maintainer to decide if this issue should be closed or not given that the upgrade is not graceful.

**Comment by mocleiri** [comment]
Sorry about causing this issue.

I looked first at what vkravets posted in relation to the 0.7 commit I pulled in but that alone was not sufficent.  

This is because the instance is already serialized and when Xstream loads it the getters are skipped.

I ended up writing a custom Converter and now the GithubSecurityRealm handles the read/write via XStream itself.

0.8 worked to read the old version but 0.8.1 was needed to actually write the new xml values.  I just deployed it and it should be available within 1 day via the jenkins plugin loader.
